network/common/src/main/java/org/apache/spark/network/server/OneForOneStreamManager.java

```diff
@@ -109,15 +111,34 @@ public void connectionTerminated(Channel channel) {
     }
   }
 
+  @Override
+  public void checkAuthorization(TransportClient client, long streamId) {
+    if (client.getClientId() != null) {
```

Comment:
I see, do you mean if the server enabled authentication, then any client that did not also enable it will fail the handshake in the first place?